### PR TITLE
CLDR-17115 Update languages/codes

### PR DESCRIPTION
**Pull request by btangmu**
-Numerous changes based on following instructions in Update Language/Script/Region Subtags

-Update world_bank_data.csv by downloading

-URL is https://databank.worldbank.org/reports.aspx?source=world-development-indicators

-Note: running AddPopulationData caused no changes

-Update supplementalData.xml by running ConvertLanguageData

-This caused removal of tok (Tok Pona) and vo (Volapük), and revision of comment for ace (Achinese to Acehnese)

CLDR-17115

- [x] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by macchiati — COMMENTED** [review]
Looks good, but remove the script.xml file, because we're addressing that in a different PR.

**Comment by btangmu** [comment]
@macchiati as you requested, I reverted scripts.xml in the last commit

**Review by macchiati — APPROVED** [review]

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt [hunk 1]
```diff
@@ -248,7 +248,7 @@ SG	;	Officially assigned	;	Singapore
 SH	;	Officially assigned	;	Saint Helena, Ascension and Tristan da Cunha
 SI	;	Officially assigned	;	Slovenia
 SJ	;	Officially assigned	;	Svalbard et l'Île Jan Mayen (le)
-SK	;	Officially assigned	;	Slovakia
+SK	;	Formerly used	;	Slovakia
```
Comment (srl295) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:251]: Why is SK 'formerly used'?

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt [hunk 2]
```diff
@@ -100,7 +100,7 @@ GA	;	Officially assigned	;	Gabon (le)
 GB	;	Officially assigned	;	United Kingdom of Great Britain and Northern Ireland (the)
 GC	;	Indeterminately reserved	;	Refers to the Patent Office of the Cooperation Council for the Arab States of the Gulf (GCC) and reserved at the request of World Intellectual Property Organization (WIPO) for use in Standard ST.3.
 GD	;	Officially assigned	;	Grenada
-GE	;	Officially assigned	;	Georgia
+GE	;	Formerly used	;	Georgia
```
Comment (srl295) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:103]: Now this seems odd..  Formerly Used??

**Review by srl295 — CHANGES_REQUESTED** [review]
ISO 3166 seems wrong

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt (hunk 1 above)
Comment (srl295) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:251]: Something's wrong here.

https://www.iso.org/obp/ui/#iso:code:3166:SK says **Officially Assigned**

Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:251]: @macchiati confusion between language/locale code for SK...

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt (hunk 2 above)
Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:103]: "The code GE was formerly used for the Gilbert and Ellice Islands (GE, GEL, --) and has been reused and reassigned to Georgia. See also code element GEKI."

I guess like SK, "Formerly used" for GE is a mistake on my part, misunderstood instructions...

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt [hunk 3]
```diff
@@ -3,7 +3,7 @@ AD	;	Officially assigned	;	Andorre (l')
 AE	;	Officially assigned	;	United Arab Emirates (the)
 AF	;	Officially assigned	;	Afghanistan
 AG	;	Officially assigned	;	Antigua and Barbuda
-AI	;	Officially assigned	;	Anguilla
+AI	;	Formerly used	;	Anguilla
```
Comment (btangmu) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:6]: Officially assigned per https://www.iso.org/obp/ui/#iso:code:3166:AI so this is another mistake?

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt [hunk 4]
```diff
@@ -30,15 +30,15 @@ BL	;	Officially assigned	;	Saint-Barthélemy
 BM	;	Officially assigned	;	Bermuda
 BN	;	Officially assigned	;	Brunei Darussalam
 BO	;	Officially assigned	;	Bolivia (Plurinational State of)
-BQ	;	Officially assigned	;	Bonaire, Sint Eustatius and Saba
+BQ	;	Formerly used	;	Bonaire, Sint Eustatius and Saba
```
Comment (btangmu) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:33]: Similar; "The code BQ was formerly used for British Antarctic Territory (BQ, ATB, --). ... BQ has been reused and reassigned to Bonaire"

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt [hunk 5]
```diff
@@ -30,15 +30,15 @@ BL	;	Officially assigned	;	Saint-Barthélemy
...
 BR	;	Officially assigned	;	Brazil
 BS	;	Officially assigned	;	Bahamas (the)
 BT	;	Officially assigned	;	Bhutan
 BU	;	Transitionally reserved	;	Burma
 BV	;	Officially assigned	;	Bouvet Island
 BW	;	Officially assigned	;	Botswana
 BX	;	Indeterminately reserved	;	Refers to Benelux Office for Intellectual Property and reserved at the request of World Intellectual Property Organization (WIPO) for use in Standard ST.3.
-BY	;	Officially assigned	;	Belarus
+BY	;	Formerly used	;	Belarus
```
Comment (btangmu) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:41]: "Officially assigned" per https://www.iso.org/obp/ui/#iso:code:3166:BY

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt [hunk 6]
```diff
@@ -70,7 +70,7 @@ DJ	;	Officially assigned	;	Djibouti
 DK	;	Officially assigned	;	Denmark
 DM	;	Officially assigned	;	Dominica
 DO	;	Officially assigned	;	Dominican Republic (the)
-DY	;	Indeterminately reserved	;	Refers to Benin (BJ, BEN, 204) and to a list of car vehicle distinguishing signs notified to the Secretary-General of the United Nations under 1949 Road Traffic Convention and 1968 Road Traffic Convention.
+DY	;	Formerly used	;	Refers to Benin (BJ, BEN, 204) and to a list of car vehicle distinguishing signs notified to the Secretary-General of the United Nations under 1949 Road Traffic Convention and 1968 Road Traffic Convention.
```
Comment (btangmu) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:73]: "Indeterminately reserved";
"Formerly used" for Dahomey, hence the confusion

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt [hunk 7]
```diff
@@ -228,7 +228,7 @@ RA	;	Indeterminately reserved	;	Refers to Argentina and to the 1949 Road Traffic
 RB	;	Indeterminately reserved	;	Refers to Bolivia or to Botswana (identical codes) and to the 1949 Road Traffic Convention. Requested by Secretary-General of UN for Road Traffic Conventions.
 RC	;	Indeterminately reserved	;	Refers to China and to the 1949 Road Traffic Convention. Requested by Secretary-General of UN for Road Traffic Conventions.
 RE	;	Officially assigned	;	Réunion
-RH	;	Indeterminately reserved	;	Refers to Haiti and to a list of car vehicle distinguishing signs notified to the Secretary-General of the United Nations under 1949 Road Traffic Convention and 1968 Road Traffic Convention.
+RH	;	Formerly used	;	Refers to Haiti and to a list of car vehicle distinguishing signs notified to the Secretary-General of the United Nations under 1949 Road Traffic Convention and 1968 Road Traffic Convention.
```
Comment (btangmu) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:231]: "Formerly used" for "Southern Rhodesia", hence the confusion

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt (hunk 3 above)
Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt:6]: "Formerly used" for "French Afars and Issas"; still "Officially assigned" for Anguilla.

@macchiati @srl295 It looks like all my changes to this file are due to my naively following the list of "Formerly used" codes shown by the ISO site, not knowing that the same code may be in different categories when was formerly used for one thing and is currently used for something completely different. So, I'll revert the whole file, OK?

**Comment by btangmu** [comment]
Per discussion, I've reverted iso_3166_status.txt to main

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt (hunk 1 above)
Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt]: reverted

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt (hunk 2 above)
Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt]: reverted

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt (hunk 4 above)
Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt]: reverted

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt (hunk 5 above)
Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt]: reverted

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt [hunk 8]
```diff
@@ -55,7 +55,7 @@ CO	;	Officially assigned	;	Colombia
 CP	;	Exceptionally reserved	;	Refers to Clipperton Island and reserved at the request of International Telecommunication Union (ITU).
 CQ  ;   Exceptionally reserved  ; Sark
 CR	;	Officially assigned	;	Costa Rica
-CS	;	Transitionally reserved	;	Serbia and Montenegro
+CS	;	Formerly used	;	Serbia and Montenegro
```
[the remark this answers is not in the document]
Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt]: reverted

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt (hunk 6 above)
Comment (btangmu) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/external/iso_3166_status.txt]: reverted

**Review by macchiati — APPROVED** [review]

**Comment by macchiati** [comment]
It looks like there are a surprising number of errors. I think it is best for me to walk you through this, and you can capture these notes in the instructions.

It appears that ISO had an unexpected number of deprecations, so you're seeing more issues that we normally see.

For lines like the following:
Error:  (TestLocale.java:921)  Error: : ajp: expected "", got "Disallowed language=ajp, status=deprecated"
Error:  (TestLocale.java:927)  Error: : ajp_Arab_JO: expected "", got "Disallowed language=ajp, status=deprecated

What is happening is that likelySubtags.xml is handling languages that are now deprecated. That is to be expected, because ISO does that occasionally, but because we added a lot of SIL language data, the number may be larger each year. To fix that, go to the file an delete the line where it is handled, and delete that line, in this case:

<likelySubtag from="ajp" to="ajp_Arab_JO" origin="sil1"/>	

    TestMacrolanguages
Error:  (TestSupplementalInfo.java:1328)  Error: Macrolanguage	sa	Sanskrit	Historical

It looks like the classification changed in ISO. We still use 'sa', because the India government disagrees that it is only historical!Add to             if (language.equals("no") || language.equals("sh")) continue; // special cases

    TestCompatibility
Error:  (TestValidity.java:284)  Error: language:dzd:deprecated => regular // add to exception list (ALLOWED_UNDELETIONS) if really un-deprecated

Check the diff in the iso-639 files to verify that dzd is really de-deprecated. Then add dzd to ALLOWED_UNDELETIONS

The "ERROR:" values below in the listing all look like keyboard stuff; I don't think those are counted. I'll file a ticket for Steven to clean those up.

File: common/validity/language.xml [hunk 9]
```diff
@@ -623,33 +625,33 @@
...
 			zxx
 		</id>
-		<id type='language' idStatus='deprecated'>		<!-- 283 items -->
-			aam adp agp ais ajt~u als aoh arb asd aue ayr ayx~y azj
+		<id type='language' idStatus='deprecated'>		<!-- 296 items -->
+			aam adp agp ais ajp ajt~u als aoh arb asd aue ayr ayx~y azj
 			baz bbz bcc bcl bgm bh bhk bic bij bjd bjq bkb blg bmy bpb btb btl bxk bxr bxx byy
 			cbe cbh cca ccq cdg cjr cka cld cmk cmn cnr coy cqu cug cum cwd
-			daf dap dgo dgu dha dhd dik diq dit djl dkl drh drr drw dud duj dwl dzd
+			daf dap dgo dgu dha dhd dik diq dit djl dkl drh drr drw dud duj dwl
```
Comment (btangmu) [inline common/validity/language.xml:632]: "dzd" was removed here

Web search for "dzd deprecated" turns up this file:

https://www.iana.org/assignments/lang-subtags-templates/dzd-2023-03-17.txt

which reads as follows:

FOR ARCHIVING: Registration form for 'dzd'

---

LANGUAGE SUBTAG REGISTRATION FORM

1. Name of requester: Doug Ewell
2. E-mail address of requester: doug at ewellic.org

3. Record Requested:

Type: language
Subtag: dzd
Description: Daza

4. Intended meaning of the subtag:

5. Reference to published description of the language (book or article):

6. Any other relevant information:

This registration tracks a change made to ISO 639-3 effective
2023-01-20, adding the code element 'dzd' for Daza, which had been
retired in 2015 as non-existent. The net effect of this registration is
to remove the Deprecated value from this record.

For more information on the ISO 639-3 change, refer to:
https://iso639-3.sil.org/request/2022-027

**Comment by btangmu** [comment]
@macchiati my latest commit fixes the problems with sa (Sanskrit) and dzd (Daza). It does not fix the problems with ajp and others in this output:
```
    testLanguageTagParserIsValid {
      Error: (TestLocale.java:921) : ajp: expected "", got "Disallowed language=ajp, status=deprecated"
      Error: (TestLocale.java:927) : ajp_Arab_JO: expected "", got "Disallowed language=ajp, status=deprecated"
      Error: (TestLocale.java:921) : kgm: expected "", got "Disallowed language=kgm, status=deprecated"
      Error: (TestLocale.java:927) : kgm_Latn_BR: expected "", got "Disallowed language=kgm, status=deprecated"
      Error: (TestLocale.java:921) : ksa: expected "", got "Disallowed language=ksa, status=deprecated"
      Error: (TestLocale.java:927) : ksa_Latn_NG: expected "", got "Disallowed language=ksa, status=deprecated"
      Error: (TestLocale.java:921) : nom: expected "", got "Disallowed language=nom, status=deprecated"
      Error: (TestLocale.java:927) : nom_Latn_PE: expected "", got "Disallowed language=nom, status=deprecated"
      Error: (TestLocale.java:921) : plj: expected "", got "Disallowed language=plj, status=deprecated"
      Error: (TestLocale.java:927) : plj_Latn_NG: expected "", got "Disallowed language=plj, status=deprecated"
      Error: (TestLocale.java:921) : prp: expected "", got "Disallowed language=prp, status=deprecated"
      Error: (TestLocale.java:927) : prp_Gujr_IN: expected "", got "Disallowed language=prp, status=deprecated"
      Error: (TestLocale.java:921) : slq: expected "", got "Disallowed language=slq, status=deprecated"
      Error: (TestLocale.java:927) : slq_Arab_IR: expected "", got "Disallowed language=slq, status=deprecated"
      Error: (TestLocale.java:921) : szd: expected "", got "Disallowed language=szd, status=deprecated"
      Error: (TestLocale.java:927) : szd_Latn_MY: expected "", got "Disallowed language=szd, status=deprecated"
      Error: (TestLocale.java:921) : tmk: expected "", got "Disallowed language=tmk, status=deprecated"
      Error: (TestLocale.java:927) : tmk_Deva_NP: expected "", got "Disallowed language=tmk, status=deprecated"
      Error: (TestLocale.java:921) : xss: expected "", got "Disallowed language=xss, status=deprecated"
      Error: (TestLocale.java:927) : xss_Cyrl_RU: expected "", got "Disallowed language=xss, status=deprecated"
      Error: (TestLocale.java:921) : zkb: expected "", got "Disallowed language=zkb, status=deprecated"
      Error: (TestLocale.java:927) : zkb_Cyrl_RU: expected "", got "Disallowed language=zkb, status=deprecated"
      Error: (TestLocale.java:921) : zua: expected "", got "Disallowed language=zua, status=deprecated"
      Error: (TestLocale.java:927) : zua_Latn_NG: expected "", got "Disallowed language=zua, status=deprecated"
```
You addressed these errors in your last comment, but I still don't understand; they're different from the "sa" error.

"ajp" occurs in languageGroup.xml, languageInfo.xml, and likelySubtags.xml. Should it be deleted from languageInfo.xml, and/or likelySubtags.xml, and then should languageGroup.xml be regenerated?

**Comment by macchiati** [comment]
Here is what to do in more detail.

### Case 1, replaced by old:

Take ajp 

Look at language-subtag-registry (the diff from the old one)

You see that ajp has 2 items added:

```
Deprecated: 2023-03-17
Preferred-Value: apc
```
That means that wherever it occurs, "apc" should be substituted. **However, if you look at apc, it is not new.** So the actions are to delete it in those files where it occurs. Search the directory supplemental. You find:

```
languageGroup.xml
93: <languageGroup parent="sem">aao abh acm acq acy aeb aec agj aii ajp akk am …</languageGroup> 

languageInfo.xml
170: <languageMatch desired="ajp" supported="ar" distance="10" oneway="true"/> <!-- South Levantine Arabic --> 

likelySubtags.xml (6 matches)
2,883: <likelySubtag from="ajp" to="ajp_Arab_JO" origin="sil1"/> <!-- South Levantine Arabic ➡︎ South Levantine Arabic (Arabic, Jordan) --> 
4,461: <likelySubtag from="gra" to="gra_Deva_IN" origin="sil1"/> <!-- Rajput Garasia ➡︎ Rajput Garasia (Devanagari, India) --> 
4,462: <likelySubtag from="gra_Gujr" to="gra_Gujr_IN" origin="sil1"/> <!-- Rajput Garasia (Gujarati) ➡︎ Rajput Garasia (Gujarati, India) --> 
```

In languageGroup:
If  'apc', didn't exist in that file you would replace it. 
Since it does, you just delete it (leaving the rest of the line alone).

```
93: <languageGroup parent="sem">aao abh acm acq acy aeb aec agj aii akk am …</languageGroup> 
```

Same in languageInfo.xml and likelySubtags.xml. 'apc' exists in each, so just delete the lines.

Suppose it were in supplementalData in the territory information (it doesn't so this is just illustration!!)
```
<territory type="PS" gdp="21220000000" literacyPercent="95.3" population="4818260">	<!--Palestinian Territories-->
  <languagePopulation type="ar" populationPercent="100" officialStatus="official"/>	<!--Arabic-->
  <languagePopulation type="apc" populationPercent="87" references="R1173"/>	<!--Levantine Arabic-->
  <languagePopulation type="ajp" populationPercent="2" references="..."/>	<!--South Levantine Arabic-->
```
In that case you would combine the two figures to get:
```
  <languagePopulation type="apc" populationPercent="89" references="R1173"/>	<!--Levantine Arabic-->
```
Use your judgment: sometimes language counts are doubled for bilingual speakers, so if it adds to a crazy amount, don't add it. (These figures are 'best available', so that's ok.)

### Case 2, no preferred

In this case, just drop the lines.

### Case 3, split
```
Subtag: ksa
Description: Shuwa-Zamani
Added: 2009-07-29
Deprecated: 2023-03-17
Comments: see izm, rsw
```
Look at iso-639-3_Retirements.tab for ksa

You'll see "Split into [rsw] Rishiwa and [izm] Kizamani"

Take the first one, and treat this case like Case 1.

**Comment by macchiati** [comment]
Right, we disabled the tool for now. It should be easy to regex-search for
(ajp|...) to find all the lines, although you want to look at each one
rather than automatically deleting.

**Comment by btangmu** [comment]
@macchiati FYI you wrote that iso-639-3_Retirements.tab says "Split into [rsw] Rishiwa and [izm] Kizamani" but the version I'm seeing (in the branch for this ticket) doesn't say anything like that -- because that file is changed in this PR! So I need to look at the version of that file before this PR. Just something to be aware of when we update the instructions...

**Comment by macchiati** [comment]
Right. What I do is look at the diffs in the PR.

BTW, as you go through this, please jot down in a doc or text file what you
are doing, so that we can use that as a basis for updating the instructions.

**Comment by btangmu** [comment]
@macchiati these two files disagree on the replacement for prp (Parsi), whether to change to gu or guj:

language-subtag-registry:
Type: language
Subtag: prp
Description: Parsi
Added: 2009-07-29
Deprecated: 2023-03-17
Preferred-Value: gu

iso-639-3_Retirements.tab:
prp	Parsi	M	guj		2023-01-20

Since your comments mainly refer to language-subtag-registry I'm guessing "gu", but it's just a wild guess so please confirm or correct!

**Comment by DavidLRowe** [comment]
I think it should be "gu". "guj" is the ISO 639-3 equivalent of "gu". The ISO 639-1 (two-letter) code is preferred if it exists.

**Comment by macchiati** [comment]
gu is the right choice. (guj is the 3 letter code, but the BCP47 uses 2
letter whenever it exists)

**Comment by btangmu** [comment]
Another disagreement, for szd -- replace with uki or umi?

iso-639-3_Retirements.tab:
szd	Seru	M	uki		2023-01-20

language-subtag-registry:
Type: language
Subtag: szd
Description: Seru
Added: 2009-07-29
Deprecated: 2023-03-17
Preferred-Value: umi

likelySubtags.xml has only umi, not uki; I'm just deleting szd from that file

**Comment by macchiati** [comment]
When in doubt, go by the language subtag registry

**Comment by btangmu** [comment]
I made another commit. Locally there's a new set of errors, which I'll work on next:

```
    TestLstrConsistency {
      Error: (TestValidity.java:537) Missing aliases for supplementalMetadata: 10
<languageAlias type="ajp" replacement="apc" reason="deprecated"/> <!-- South Levantine Arabic ⇒ Levantine Arabic -->
<languageAlias type="kgm" replacement="plu" reason="deprecated"/> <!-- Karipúna ⇒ Palikúr -->
<languageAlias type="nom" replacement="cbr" reason="deprecated"/> <!-- Nocamán ⇒ Cashibo-Cacataibo -->
<languageAlias type="pmk" replacement="crr" reason="deprecated"/> <!-- Pamlico ⇒ Carolina Algonquian -->
<languageAlias type="prp" replacement="gu" reason="deprecated"/> <!-- Parsi ⇒ Gujarati -->
<languageAlias type="szd" replacement="umi" reason="deprecated"/> <!-- Seru ⇒ Ukit -->
<languageAlias type="tmk" replacement="tdg" reason="deprecated"/> <!-- Northwestern Tamang ⇒ Western Tamang -->
<languageAlias type="tpw" replacement="tpn" reason="deprecated"/> <!-- Tupí ⇒ Tupinambá -->
<languageAlias type="xss" replacement="zko" reason="deprecated"/> <!-- Assan ⇒ Kott -->
<languageAlias type="zkb" replacement="kjh" reason="deprecated"/> <!-- Koibal ⇒ Khakas -->
```

**Comment by macchiati** [comment]
Good of it to tell you exactly which lines to add!

**Comment by btangmu** [comment]
> Good of it to tell you exactly which lines to add!

Add where?

**Comment by btangmu** [comment]
Tests are passing!

File: common/validity/language.xml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <?xml version='1.0' encoding='UTF-8' ?>
 <!DOCTYPE supplementalData SYSTEM '../../common/dtd/ldmlSupplemental.dtd'>
 <!--
-	Copyright © 1991-2022 Unicode, Inc.
+	Copyright © 1991-2024 Unicode, Inc.
 	For terms of use, see http://www.unicode.org/copyright.html
 	SPDX-License-Identifier: Unicode-DFS-2016
```
Comment (srl295) [inline common/validity/language.xml:6]: Should be `Unicode-3.0`

Comment (srl295) [reply common/validity/language.xml:6]: I'll do another pass replacing identifiers.

Comment (btangmu) [reply common/validity/language.xml:6]: Thanks!

**Review by macchiati — APPROVED** [review]

**Pull request merged**